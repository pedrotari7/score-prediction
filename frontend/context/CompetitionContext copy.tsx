import { createContext } from 'react';
import { GroupMap } from '../../interfaces/main';

const GroupMapContext = createContext<GroupMap>({});

export default GroupMapContext;
