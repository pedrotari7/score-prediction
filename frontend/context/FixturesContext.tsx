import { createContext } from 'react';
import { Fixtures } from '../../interfaces/main';

const FixturesContext = createContext<Fixtures | null>(null);

export default FixturesContext;
