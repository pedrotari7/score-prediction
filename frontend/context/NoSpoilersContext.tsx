import { createContext } from 'react';

const NoSpoilersContext = createContext<{ noSpoilers: boolean | null; setNoSpoilers: (ns: boolean) => void } | null>(
	null
);

export default NoSpoilersContext;
