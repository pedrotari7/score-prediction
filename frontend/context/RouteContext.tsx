import { createContext } from 'react';

export enum Route {
	Home = 'home',
	Predictions = 'predictions',
	Standings = 'standings',
	Leaderboard = 'leaderboard',
	ListLeaderboards = 'listLeaderboards',
	Settings = 'settings',
	Match = 'match',
	Rules = 'rules',
	Users = 'users',
	JoinLeaderboard = 'joinLeaderboard',
	RefreshPage = 'refreshPage',
}

export interface RouteInfo {
	page: Route;
	data?: string | number | undefined;
}

const RouteContext = createContext<{ route: RouteInfo; setRoute: (r: RouteInfo) => void } | null>(null);

export default RouteContext;
