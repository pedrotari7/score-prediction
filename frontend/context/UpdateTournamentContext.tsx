import { createContext } from 'react';

const UpdateTournamentContext = createContext<(() => Promise<void>) | null>(null);

export default UpdateTournamentContext;
