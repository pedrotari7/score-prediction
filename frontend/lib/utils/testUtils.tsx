import { render } from '@testing-library/react';
import { ReactNode, ContextType } from 'react';
import RouteContext from '../../context/RouteContext';
import NoSpoilersContext from '../../context/NoSpoilersContext';

export const customRender = (
	ui: ReactNode,
	options?: {
		routeContextProps?: ContextType<typeof RouteContext>;
		noSpoilersContextProps?: ContextType<typeof NoSpoilersContext>;
		renderOptions?: Parameters<typeof render>[1];
	}
) => {
	return render(
		<RouteContext.Provider value={options?.routeContextProps ?? null}>
			<NoSpoilersContext.Provider value={options?.noSpoilersContextProps ?? null}>
				{ui}
			</NoSpoilersContext.Provider>
		</RouteContext.Provider>,
		options?.renderOptions
	);
};
