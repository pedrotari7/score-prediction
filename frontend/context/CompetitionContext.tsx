import { createContext } from 'react';
import { Competition } from '../../interfaces/main';
import { competitions } from '../../shared/utils';

const CompetitionContext = createContext<Competition>(competitions.euro2024);

export default CompetitionContext;
