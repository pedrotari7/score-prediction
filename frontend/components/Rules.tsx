import { useContext } from 'react';
import CompetitionContext from '../context/CompetitionContext';
import { classNames, getCompetitionClass } from '../lib/utils/reactHelper';

const Rules = () => {
	const competition = useContext(CompetitionContext);
	const gcc = (p: string) => getCompetitionClass(p, competition);

	return (
		<div
			className={classNames(
				gcc('text-light'),
				gcc('bg-dark'),
				`flex flex-col justify-center select-none  m-8 p-8 shadow-pop rounded-md`,
				'mx-8 md:mx-24 lg:mx-48'
			)}>
			<div className="font-bold mb-4 text-4xl">Rules</div>
			<div className="font-bold text-2xl mb-4">Points in each game</div>
			<div className="rounded-md p-2 mb-4 bg-green-500 flex flex-row flex-wrap text-lg">
				Exact Score (right outcome of the game and exact number of goals scored by both teams)
				<span className="font-bold text-2xl mt-2 w-full">+5 Points</span>
			</div>
			<div className="rounded-md p-2 mb-4 bg-yellow-500 flex flex-row flex-wrap text-lg">
				Correct Result (right outcome of the game, but wrong number of goals of at least one of the teams)
				<span className="font-bold text-2xl mt-2 w-full">+3 Points</span>
			</div>
			<div className="rounded-md p-2 mb-4 bg-pink-500 flex flex-row flex-wrap text-lg">
				Right number of goals from one of the teams
				<span className="font-bold text-2xl mt-2 w-full">+1 Point</span>
			</div>
			<div className="rounded-md p-2 mb-4 bg-red-500 flex flex-row flex-wrap text-lg">
				None of the above
				<span className="font-bold text-2xl mt-2 w-full">+0 Points</span>
			</div>
			<div className="mb-4">
				You are only awarded points once per game, so from the bullet points above you will receive the one that
				awards you the highest number of points. Each game will be colored with the respective point color.
			</div>
			<div>
				<div className="font-bold text-2xl mb-4">Groups</div>
				<div className="bg-purple-700 rounded-md p-2 mb-4 flex flex-row flex-wrap text-lg">
					You will be awarded 1 point per each succesful guess on a team&apos;s final position in the group
					stage.
					<span className="font-bold text-2xl mt-2 w-full">+1 Point</span>
				</div>
			</div>
			<div className="mb-4">
				<div className="font-bold text-2xl mb-4">Final Stages</div>
				In the final stages of the tournament each game is similar to the group stage games, but in this case
				your prediction will be compared with the result at the end of the 90 or 120 minutes.
				<div className="bg- bg-gray-500 rounded-md p-2 my-4 ">
					Nonetheless, in case of penalty shootout if you guessed correctly the team that ends up going
					through to the next stage, you will be awarded 1 extra point that adds to the ones that you might
					have acheived during the regular time.
				</div>
			</div>
			<div className="">
				<div className="font-bold text-2xl mb-4">Deadlines</div>
				In the group stages the predictions for each game must be done before its start. For the final stages
				games, the predictions will be available once the matchup is known and have to be submited before the
				game starts.
			</div>
		</div>
	);
};

export default Rules;
