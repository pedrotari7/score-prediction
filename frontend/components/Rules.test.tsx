import Rules from './Rules';
import { Route } from '../context/RouteContext';
import { customRender } from '../lib/utils/testUtils';

describe('Rules', () => {
	it('renders empty page without route info', () => {
		const { container } = customRender(<Rules />);
		expect(container).toMatchSnapshot();
	});

	it('render rules page', () => {
		const { container } = customRender(<Rules />, {
			routeContextProps: { setRoute: () => {}, route: { page: Route.Rules } },
		});
		expect(container).toMatchSnapshot();
	});

	it('click my predictions should change route', () => {
		const routeContextProps = { setRoute: jest.fn(), route: { page: Route.Rules, data: 'some data' } };
		const { container } = customRender(<Rules />, { routeContextProps });

		const button = container.querySelector('[test-id=my-predictions-button]') as HTMLDivElement;

		button.click();

		expect(routeContextProps.setRoute).toHaveBeenCalledWith({
			page: Route.Predictions,
			data: routeContextProps.route.data,
		});
	});
});
