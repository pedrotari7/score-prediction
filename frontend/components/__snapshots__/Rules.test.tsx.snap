// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rules render rules page 1`] = `
<div>
  <div
    class="text-light-euro2024 bg-dark-euro2024 m-8 flex select-none flex-col justify-center rounded-md p-8 shadow-pop mx-8 md:mx-24 lg:mx-48"
  >
    <div
      class="mb-4 text-4xl font-bold"
    >
      Rules
    </div>
    <div
      class="mb-4 text-2xl font-bold"
    >
      Points in each game
    </div>
    <div
      class="mb-4 flex flex-row flex-wrap rounded-md bg-green-600 p-2 text-lg"
    >
      Exact Score (right outcome of the game and exact number of goals scored by both teams)
      <span
        class="mt-2 w-full text-2xl font-bold"
      >
        +
        3
         Points
      </span>
    </div>
    <div
      class="mb-4 flex flex-row flex-wrap rounded-md bg-yellow-600 p-2 text-lg"
    >
      Correct Result (right outcome of the game, but wrong number of goals of at least one of the teams)
      <span
        class="mt-2 w-full text-2xl font-bold"
      >
        +
        2
         Points
      </span>
    </div>
    <div
      class="mb-4 flex flex-row flex-wrap rounded-md bg-pink-600 p-2 text-lg"
    >
      Right number of goals from one of the teams
      <span
        class="mt-2 w-full text-2xl font-bold"
      >
        +
        1
         Point
      </span>
    </div>
    <div
      class="mb-4 flex flex-row flex-wrap rounded-md bg-red-600 p-2 text-lg"
    >
      None of the above
      <span
        class="mt-2 w-full text-2xl font-bold"
      >
        +0 Points
      </span>
    </div>
    <div
      class="mb-4"
    >
      You are only awarded points once per game, so from the bullet points above you will receive the one that awards you the highest number of points. Each game will be colored with the respective point color.
    </div>
    <div>
      <div
        class="mb-4 text-2xl font-bold"
      >
        Groups
      </div>
      <div
        class="mb-4 flex flex-row flex-wrap rounded-md bg-purple-700 p-2 text-lg"
      >
        You will be awarded 1 point per each successful guess on a team's final position in the group stage.
        <span
          class="mt-2 w-full text-2xl font-bold"
        >
          +
          1
           Point
        </span>
      </div>
    </div>
    <div
      class="mb-4"
    >
      <div
        class="mb-4 text-2xl font-bold"
      >
        Final Stages
      </div>
      In the final stages of the tournament each game is similar to the group stage games, but in this case your prediction will be compared with the result at the end of the 90 or 120 minutes.
      <div
        class="my-4 rounded-md bg-gray-500 p-2"
      >
        Nonetheless, in case of penalty shootout if you guessed correctly the team that ends up going through to the next stage, you will be awarded
         
        <span
          class="font-bold"
        >
          1
           extra point
        </span>
         that adds to the ones that you might have achieved during the regular time.
      </div>
    </div>
    <div
      class=""
    >
      <div
        class="mb-4 text-2xl font-bold"
      >
        Deadlines
      </div>
      In the group stages the predictions for each game must be done before their start. For the final stages games, the predictions will be available once the matchup is known and have to be submitted before the game starts.
    </div>
  </div>
  <div
    class="text-light-euro2024 m-8 flex select-none flex-col justify-center rounded-md p-8 mx-8 mb-10 md:mx-24 lg:mx-48"
  />
  <div
    class="bg-light-euro2024 fixed bottom-4 right-4 cursor-pointer rounded-md p-4 font-bold shadow-pop"
    test-id="my-predictions-button"
  >
    My Predictions
  </div>
</div>
`;

exports[`Rules renders empty page without route info 1`] = `<div />`;
